fontforge/macbinary.c

```diff
@@ -2106,15 +2106,9 @@ static SplineFont *SearchTtfResources(FILE *f,long rlistpos,int subcnt,long rdat
 return( (SplineFont *) names );
 	}
 	if ((pt = strrchr(filename,'/'))==NULL ) pt = filename;
-	/* Someone gave me a font "Nafees Nastaleeq(Updated).ttf" and complained */
-	/*  that ff wouldn't open it */
-	/* Now someone will complain about "Nafees(Updated).ttc(fo(ob)ar)" */
-	if ( (lparen = strrchr(pt,'('))!=NULL &&
-		(rparen = strrchr(lparen,')'))!=NULL &&
-		rparen[1]=='\0' ) {
-	    char *find = copy(lparen+1);
-	    pt = strchr(find,')');
-	    if ( pt!=NULL ) *pt='\0';
+	if ( lparen = SFSubfontnameStart(pt) ) {
+	    find = copy(lparen+1);
+	    find[strlen(find)-1] = '\0';
```

Comment:
It looks like SFSubfontnameStart always returns `NULL` or a string of length greater than zero. Is that the basis of this assumption? If so, I would put that in a comment so that nobody worries about a segfault on `""`.

Reply:
Added comment about this for `SFSubfontnameStart`

Reply:
When you create code, take a look at what is compiled, and not simply looking for an X or a checkmark saying it passes.... for example, if you follow the github checkmark and look at what was compiled, you'll see this should be avoided:
```code
  if ( lparen = SFSubfontnameStart(pt) ) {
       ^~~~~~
2574:10: warning: suggest parentheses around assignment used as truth value [-Wparentheses]
```